pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/GenericTransformFunctionTest.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.common.request.transform.TransformExpressionTree;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class GenericTransformFunctionTest extends BaseTransformFunctionTest {
+
+  @Test
```

Comment:
I believe this exercises the compilation path. You should add tests in CalciteSqlCompiler test file as well. 

Also, we should add tests for end to end query execution. See TransformQueriesTest class or consider adding these to one of the existing query tests